docker-compose-full-services.yml

```
@@ -0,0 +1,130 @@
+  #    (ecto_sql) lib/mix/tasks/ecto.migrate.ex:106: Mix.Tasks.Ecto.Migrate.run/2\n    (mix) lib/mix/task.ex:331:
+  # Mix.Task.run_task/3\n    (mix) lib/mix/cli.ex:79: Mix.CLI.run_task/2\n    (elixir) lib/code.ex:767:
+  # Code.require_file/2\n'
+  #
+  # watcher_1            | 2019-06-19 10:21:05,641 CRITICAL:Could not update the Postgres database Exiting.
+  #
+  postgres:
+    image: postgres
+    restart: always
+    ports:
+      - "5433:5432"
+    environment:
+      POSTGRES_USER: omisego_dev
+      POSTGRES_PASSWORD: omisego_dev
+      POSTGRES_DB: omisego_dev
+    healthcheck:
+      test: pg_isready -U omisego_dev
+      interval: 5s
+      timeout: 3s
+      retries: 5
+
+  geth:
+    image: ethereum/client-go:v1.8.21
+    entrypoint: /bin/sh -c "apk add curl && geth --targetgaslimit "6200000" --dev --dev.period 1 --rpc --rpcapi personal,web3,eth,net --rpcaddr 0.0.0.0 --rpcvhosts=* --rpcport=8545"
+    ports:
+     - "8545:8545"
+    healthcheck:
+      test: curl geth:8545
+      interval: 5s
+      timeout: 3s
+      retries: 5
+
+  # TODO(jacob): this exists in the mac version of the file, but not the non-mac one. understand why and
+  # check whether this should be in this file or not.
+  elixir-omg:
+    build: .
+    image: elixir-omg:dockercompose
+
+  contractexchanger:
+    image: jakebunce/contractexchanger:v0.0.2
+    ports:
+      - "5000:5000"
+    healthcheck:
+      test: curl contractexchanger:5000
+      interval: 5s
+      timeout: 3s
+      retries: 5
+    depends_on:
+      geth:
+        condition: service_healthy
+
+  childchain:
+    image: elixir-omg:dockercompose
+    entrypoint: /bin/bash -c "./launcher.py && elixir --erl '-sname childchain' -S mix xomg.child_chain.start --config ~/config.exs"
+    environment:
+      - MIX_ENV=dev
+      - EXIT_PERIOD_SECONDS=86400
+      - ELIXIR_SERVICE=CHILDCHAIN
+      - CONTRACT_EXCHANGER_URL=http://contractexchanger:5000
+      - ETHEREUM_RPC_URL=http://geth:8545
+      - CHILD_CHAIN_URL=http://childchain:9656
+      - ETHEREUM_NETWORK=LOCALCHAIN
+    restart: always
+    ports:
+      - "9656:9656"
+    healthcheck:
+      test: curl childchain:9656
+      interval: 10s
+      timeout: 3s
+      retries: 7
+    depends_on:
+      geth:
+        condition: service_healthy
+      contractexchanger:
+        condition: service_healthy
+
+  watcher:
+    image: elixir-omg:dockercompose
+    entrypoint: /bin/bash -c "./launcher.py && elixir --erl '-sname watcher' -S mix xomg.watcher.start --convenience --config ~/config_watcher.exs "
+    environment:
+      - MIX_ENV=dev
+      - ELIXIR_SERVICE=WATCHER
+      - CONTRACT_EXCHANGER_URL=http://contractexchanger:5000
+      - ETHEREUM_RPC_URL=http://geth:8545
+      - CHILD_CHAIN_URL=http://childchain:9656
+      - ETHEREUM_NETWORK=LOCALCHAIN
+      - DATABASE_URL=postgres://omisego_dev:omisego_dev@postgres:5433/omisego_dev
```

I *think* this is because we are calling the service by name directly? I just tried and it looks like changing the port to `5432` fixed it.

```suggestion
      - DATABASE_URL=postgres://omisego_dev:omisego_dev@postgres:5432/omisego_dev
```

---

Ah, that's great. Weird though because postgres service is defined to map port 5432 to 5433. The error I saw wasn't that it couldn't connect to postgres, but that there was a DB migration problem.